### DISCUSSION TITLE
Handling very large (50MB) source files

### DISCUSSION THREAD

**Discussion by chkp-omeran** [dotnet/roslyn#66434] [Q&A]
I'm trying to use Roslyn for modifying serializers code generated by svcutil.exe. (basically, it contains several dozens medium classes, and two huge classes (reader and writer) that contains a function for every class in a very large schema).
It works great for medium files, but when trying to parse a very large source file (50MB, ~600K lines), I'm getting an empty 'Members' collection for the CompilationUnitRoot, without any error.
Is there any way around this?

Edit - apparently, this happens due a monstrous (11K lines) if/else statement that breaks the stack since it's parsed in recursion. I'll try to parse it separately and extract it from the file.

**Comment by CyrusNajmabadi** [comment]
Can you provide a repro?

**Reply by chkp-omeran** [reply]
It's a bit problematic, because this code includes all of our schema structure. 
Anyway I was able to avoid it by pre-processing the file and replace all "return; } else if" with "return; } if".

**Reply by Xyncgas** [reply]
If Microsoft cares about the quality of their product, might wanna investigate this even without a repo, because a legitimate user has voiced a legitimate concern. Let me help you by posting this to my twitter, hopefully someone in microsoft would take a look

**Reply by CyrusNajmabadi** [reply]
@Xyncgas the issue was figured out. It was a case of extremely nested if-statements.
